src/jsdoc_transformer.ts

```diff
@@ -615,7 +616,19 @@ export function jsdocTransformer(
               // getOriginalNode(decl) is required because the type checker cannot type check
               // synthesized nodes.
               const typeStr = moduleTypeTranslator.typeToClosure(ts.getOriginalNode(decl));
-              localTags.push({tagName: 'type', type: typeStr});
+              // If @define is present then do not add @type, but instead verify that any type is
+              // correct. Specifying any type at all here may become an error later.
+              const defineTag = localTags.find(({tagName}) => tagName === 'define');
+              if (!defineTag) {
+                localTags.push({tagName: 'type', type: typeStr});
+              } else if (defineTag.type !== typeStr) {
+                if (defineTag.type !== undefined) {
+                  moduleTypeTranslator.error(
+                      varStmt,
+                      `incorrect type in @define: found '${defineTag.type}' required '${typeStr}'`);
```

Comment:
Since one of those is inferred the comment can be confusing. I would add some more text like:

'required' -> 'TypeScript type was'
'found' -> 'written Closure type was'

Otherwise it looks too much like a TS type checking error, which it is not.

Reply:
Done.